packages/suite/src/components/guide/GuideMarkdown.tsx

```diff
@@ -93,6 +111,31 @@ export const GuideMarkdown = ({ markdown }: GuideMarkdownProps) => {
                                 </TrezorLink>
                             );
                         },
+                        blockquote: ({ children }) => {
```

Comment:
I would create a new component `GuideBlockquote`

something like
```
const StyledWarning = here..

const BULB_EMOJI = '💡';
const WARNING_EMOJI = '⚠️';
const regex = new RegExp(`^(${BULB_EMOJI}|${WARNING_EMOJI})\\s*`);

const GuideBlockquote = ({ children }: GuideBlockquoteProps) => {
    const message = children?.[1]?.props?.children;

    if (!message) {
        console.error(`Unexpected variant of Guide hint.`); // Q: not sure if string is correct
        return null; // Q: if no message let's show nothing? instead of logging only an error
    }

    const emoji = message[0];
    message[0] = message[0]?.replace(regex, ''); Q: why do you modify message, but you use {children} in <StyledWarning>

    let variant = "learn";
    switch(emoji) {
         case BULB_EMOJI:
              variant = "learn";
              break;
         case WARNING_EMOJI:
         default:
              variant = "warning";
               break;
    }

    return (
        <StyledWarning withIcon variant={variant}>
            {children}
        </StyledWarning>
    );
};
```

Reply:
Refactored.

To your questions:
- `not sure if string is correct`: I think it should be OK, also that's how it is handled with the link component and missing `href`
- `if no message let's show nothing? instead of logging only an error`: If we cannot parse the message for some reason, `children` should still be able to display, maybe it is just a string and no parsing is needed... but I am not sure on this one.
-  `why do you modify message, but you use {children} in <StyledWarning>`: It is a reference to an array within that object so it does affect the object. But it is confusing so I added a comment.